qiskit_nature/transformers/second_quantization/electronic/active_space_transformer.py

```diff
@@ -367,24 +367,13 @@ def _transform_property(self, prop: Property) -> Property:
         if isinstance(prop, GroupedProperty):
             transformed_property = deepcopy(prop)
 
-            # Get the iterator of the Group's properties. We access __iter__() directly to make
-            # mypy happy :-)
-            iterator = transformed_property.__iter__()
-
-            transformed_internal_property = None
-            while True:
-                try:
-                    # Send the transformed internal property to the GroupedProperty generator.
-                    # NOTE: in the first iteration, this variable is None, which is equivalent to
-                    # starting the iterator.
-                    # NOTE: a Generator's send method returns the iterators next value [2].
-                    # [2]: https://docs.python.org/3/reference/expressions.html#generator.send
-                    internal_property = iterator.send(transformed_internal_property)
-                except StopIteration:
-                    break
-
+            for internal_property in iter(prop):
                 try:
                     transformed_internal_property = self._transform_property(internal_property)
+                    if transformed_internal_property is not None:
+                        transformed_property.add_property(transformed_internal_property)
+                    else:
+                        transformed_property.remove_property(internal_property)
```

Comment:
A short FYI comment from my side:
The initial implementation was trying to be clever by using `generator.send` to inject the newly transformed property back into the iterated object. However, this does not work as intended when you want to remove properties because sending `None` will not overwrite the existing iterated object.